examples/endian_swapper/cosim/Makefile

```diff
@@ -1,30 +1,32 @@
-include $(shell cocotb-config --makefiles)/Makefile.inc
-include $(shell cocotb-config --makefiles)/Makefile.pylib
+PYTHON_BIN ?= $(realpath $(shell cocotb-config --python-bin))
 
-PYTHON_LIBDIR ?= /usr/lib64
-SWIG ?= swig
-CC=gcc
-ifeq ($(ARCH),i686)
-	CC += -m32
+PYTHON_LIBDIR := $(shell $(PYTHON_BIN) -c 'from distutils import sysconfig; print( sysconfig.get_config_var("LIBDIR") )')
+ifeq ($(PYTHON_LIBDIR),None)
+    PYTHON_LIBDIR := $(shell dirname $(PYTHON_BIN))/libs
 endif
 
-ifeq ($(OS),Darwin)
-PYTHON_LD_FLAGS += -undefined dynamic_lookup
+PYTHON_INCLUDEDIR := $(shell $(PYTHON_BIN) -c 'import distutils.sysconfig; print( distutils.sysconfig.get_python_inc() )')
+
+SWIG ?= swig
+CC ?= gcc
+
+ifeq ($(shell uname),Darwin)
+    CFLAGS += -undefined dynamic_lookup
 endif
 
 .PHONY: all
 all: io_module.so _hal.so
 
 io_module.o: io.c io_module.h io.h
-	$(CC) $(GCC_FLAGS) -fPIC -pthread -fno-strict-aliasing -O2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 \
+	$(CC) $(CFLAGS) -fPIC -pthread -fno-strict-aliasing -O2 -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 \
 	-fexceptions --param=ssp-buffer-size=4 -D_GNU_SOURCE \
 	-fwrapv -DNDEBUG -I$(PYTHON_INCLUDEDIR) -c $< -o $@
```

Comment:
While you're at it: can you remove the NDEBUG (ideally in a separate commit)? It seems a dangerous choice for a piece of demo software.

Reply:
OK. Can do separate.

Reply:
https://github.com/cocotb/cocotb/pull/1621

Reply:
Again, this might be easier to just let setuptools handle